tests/unit_tests/data_steward/tools/purge_hpo_data_test.py

```diff
@@ -0,0 +1,98 @@
+import unittest
+from unittest import mock
+
+from google.cloud.bigquery.table import TableListItem, TableReference
+from google.cloud.bigquery import Client, DatasetReference
+
+import bq_utils
+from resources import CDM_TABLES
+from tools.purge_hpo_data import purge_hpo_data, _filter_hpo_tables
+
+
+class PurgeHpoDataTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print(
+            '\n**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'fake_project'
+        self.dataset_id = 'fake_dataset'
+        self.dataset = DatasetReference(self.project_id, self.dataset_id)
+        self.hpo_ids = ['fake_hpo1', 'fake_hpo2']
+        self.hpo_cdm_tables = [
+            self._table_list_item(hpo_id, cdm_table)
+            for hpo_id in self.hpo_ids
+            for cdm_table in CDM_TABLES
+        ]
+        self.hpo_cdm_and_achilles_tables = [
+            self._table_list_item(hpo_id, cdm_table)
+            for hpo_id in self.hpo_ids
+            for cdm_table in CDM_TABLES + ['achilles', 'achilles_results']
+        ]
+
+    def _full_table_id(self, table_id):
+        return f'{self.project_id}.{self.dataset_id}.{table_id}'
+
+    def _table_id_to_list_item(self, table_id):
+        full_table_id = self._full_table_id(table_id)
+        resource = {
+            "tableReference":
+                TableReference.from_string(full_table_id).to_api_repr()
+        }
+        return TableListItem(resource)
+
+    def _table_list_item(self, hpo_id, cdm_table) -> TableListItem:
+        table_id = bq_utils.get_table_id(hpo_id, cdm_table)
+        return self._table_id_to_list_item(table_id)
+
+    def test_filter_hpo_tables(self):
+        # clinical tables for specified hpo are returned
+        results = _filter_hpo_tables(self.hpo_cdm_tables, 'fake_hpo1')
+        expected_ids = [
+            bq_utils.get_table_id('fake_hpo1', cdm_table)
+            for cdm_table in CDM_TABLES
+        ]
+        actual_ids = [table.table_id for table in results]
+        self.assertListEqual(expected_ids, actual_ids)
+
+        # non-clinical tables are NOT returned
+        results = _filter_hpo_tables(self.hpo_cdm_and_achilles_tables,
+                                     'fake_hpo1')
+        actual_ids = [table.table_id for table in results]
+        self.assertListEqual(expected_ids, actual_ids)
+
+        # empty list if no tables associated with hpo_id found
+        results = _filter_hpo_tables(self.hpo_cdm_tables, 'fake_hpo3')
+        self.assertListEqual([], results)
+
+    def _mock_client_with(self, tables):
+        """
+        Get a mock BQ client object
+        :param tables: names of tables the client should list
+        :return: the mock client object
+        """
+        mock_client = mock.MagicMock(wraps=Client(project=self.project_id))
```

Comment:
There is something wrong with this mock client instantiation.  It took me a little bit to reproduce the error, but you can reproduce it locally if you run:
```
unset GOOGLE_APPLICATION_CREDENTIALS
cd curation
./tests/run_tests.sh -s unit -r purge_hpo*
```

I tried creating a mock credentials object, but that didn't work well with the wrapped client object.  The only thing I was able to do locally after unsetting the `GOOGLE_APPLICATION_CREDENTIALS` variable was
```suggestion
        mock_client = mock.MagicMock()
```
I tried creating a credentials object with bad data, but it ran into errors in the pyasn1 module and I didn't want to look deeper than that.

Reply:
As discussed, I changed it to a plain ol' `MagicMock` minus the `wraps` bit